crates/rolldown_resolver/src/resolver.rs

```diff
@@ -121,6 +121,8 @@ impl<F: FileSystem + Default> Resolver<F> {
       symlinks: raw_resolve.symlinks.unwrap_or(true),
       builtin_modules,
       module_type: true,
+      // This is not part of the spec, but required to align with vite.
+      allow_package_exports_in_directory_resolve: true,
```

Comment:
Should we enable this by default? We should only enable this when `experimental#vite_mode` is true.

Or `rolldown-vite` is using `rolldown_plugin_vite_resolve`, which should enable this in that part of code.

@sapphi-red cc

Reply:
I don't know :-) You guys need to configure this correctly for ecosystem to work correctly 😅

Reply:
Pushed a commit to only set this in the vite resolve plugin 👍